ipc/cli/src/commands/subnet/set_federated_power.rs

```diff
@@ -0,0 +1,52 @@
+use std::str::FromStr;
+use async_trait::async_trait;
+use clap::Args;
+use fvm_shared::address::Address;
+use ipc_api::subnet_id::SubnetID;
+use crate::{CommandLineHandler, GlobalArguments};
+use crate::commands::{get_ipc_provider, require_fil_addr_from_str};
+
+/// The command to set federated power.
+pub struct SetFederatedPower;
+
+#[async_trait]
+impl CommandLineHandler for crate::commands::subnet::SetFederatedPower {
+    type Arguments = crate::commands::subnet::SetFederatedPowerArgs;
+
+    async fn handle(global: &GlobalArguments, arguments: &Self::Arguments) -> anyhow::Result<()> {
+        log::debug!("set federated power with args: {:?}", arguments);
+
+        let provider = get_ipc_provider(global)?;
+        let subnet = SubnetID::from_str(&arguments.subnet)?;
+
+        let addresses: Vec<Address> = arguments.validator_addresses.iter().map(
+            |address| require_fil_addr_from_str(address).unwrap()
+        ).collect();
+
+        let public_keys: Vec<Vec<u8>> = arguments.validator_pubkeys.iter().map(
+            |key| hex::decode(key).unwrap()
+        ).collect();
+
+        let from_address = require_fil_addr_from_str(&arguments.from).unwrap();
+
+        let chain_epoch = provider.set_federated_power(&from_address, &subnet, &addresses, &public_keys, &arguments.validator_power).await?;
+        println!("New federated power is set at epoch {chain_epoch}");
+
+        Ok(())
+    }
+}
+
+#[derive(Debug, Args)]
+#[command(name = "stake", about = "Set federated power for validators")]
```

Comment:
The command name should be `set-federated-power`. 
btw `stake` is already taken by other command so IIUC this command is not visible to the user - were you able to test it eg on Calibration?